homeassistant/components/rainmachine/config_flow.py

```diff
@@ -0,0 +1,88 @@
+"""Config flow to configure the RainMachine component."""
+
+from collections import OrderedDict
+
+import voluptuous as vol
+
+from homeassistant import config_entries
+from homeassistant.core import callback
+from homeassistant.const import (
+    CONF_IP_ADDRESS, CONF_PASSWORD, CONF_PORT, CONF_SCAN_INTERVAL, CONF_SSL)
+from homeassistant.helpers import aiohttp_client
+
+from .const import DEFAULT_PORT, DEFAULT_SCAN_INTERVAL, DEFAULT_SSL, DOMAIN
+
+
+@callback
+def configured_instances(hass):
+    """Return a set of configured RainMachine instances."""
+    return set(
+        entry.data[CONF_IP_ADDRESS]
+        for entry in hass.config_entries.async_entries(DOMAIN))
+
+
+@config_entries.HANDLERS.register(DOMAIN)
+class RainMachineFlowHandler(config_entries.ConfigFlow):
+    """Handle a RainMachine config flow."""
+
+    VERSION = 1
+    CONNECTION_CLASS = config_entries.CONN_CLASS_LOCAL_POLL
+
+    def __init__(self):
+        """Initialize the config flow."""
+        self.data_schema = OrderedDict()
+        self.data_schema[vol.Required(CONF_IP_ADDRESS)] = str
+        self.data_schema[vol.Required(CONF_PASSWORD)] = str
+        self.data_schema[vol.Optional(CONF_PORT, default=DEFAULT_PORT)] = int
+        self.data_schema[vol.Optional(CONF_SSL, default=True)] = bool
```

Comment:
Is this option too advanced for config flow?

Reply:
I don't think so. From my research, some people choose to put self-signed certs on their controller; others don't. Considering that variability, I'd like to suggest we leave it.

Reply:
Researching further: sounds like disabling SSL is a very uncommon thing. So, based on your feedback, I'm thinking we can lock `True` and remove this option from the config entry. 👍